daemon/mgr/network.go

```diff
@@ -235,13 +236,58 @@ func (nm *NetworkManager) EndpointCreate(ctx context.Context, endpoint *types.En
 	if epInfo.GatewayIPv6().To16() != nil {
 		endpointConfig.IPV6Gateway = epInfo.GatewayIPv6().String()
 	}
+	endpoint.ID = ep.ID()
+	endpointConfig.EndpointID = ep.ID()
+	endpointConfig.NetworkID = n.ID()
+
+	iface := epInfo.Iface()
+	if iface != nil {
+		if iface.Address() != nil {
+			mask, _ := iface.Address().Mask.Size()
+			endpointConfig.IPPrefixLen = int64(mask)
+			endpointConfig.IPAddress = iface.Address().String()
+		}
+
+		if iface.MacAddress() != nil {
+			endpointConfig.MacAddress = iface.MacAddress().String()
+		}
+	}
 
 	return endpointName, nil
 }
 
-// EndpointRemove is used to create network endpoint.
-func (nm *NetworkManager) EndpointRemove(ctx context.Context, name string) error {
-	// TODO
+// EndpointRemove is used to remove network endpoint.
+func (nm *NetworkManager) EndpointRemove(ctx context.Context, endpoint *types.Endpoint) error {
+	sid := endpoint.NetworkConfig.SandboxID
+	epConfig := endpoint.EndpointConfig
+
+	logrus.Debugf("remove endpoint: %s on network: %s", epConfig.EndpointID, endpoint.Name)
+
+	if sid == "" {
+		return nil
+	}
+
+	sb, err := nm.controller.SandboxByID(sid)
+	if err == nil {
+		if err := sb.Delete(); err != nil {
+			logrus.Errorf("failed to delete sandbox id: %s, err: %v", sid, err)
+			return err
+		}
+	} else if _, ok := err.(networktypes.NotFoundError); !ok {
+		logrus.Errorf("failed to get sandbox id: %s, err: %v", sid, err)
+		return fmt.Errorf("failed to get sandbox id: %s, err: %v", sid, err)
+	}
+
+	// clean endpoint configure data
```

Comment:
Could you encapsulate all the below assignment into a new function like `CleanupEndpointConfig` to simplify the code? And maybe a comment which represents the usage of this function is much more helpful.

Reply:
fixed